File: jput-core/src/main/java/cz/encircled/jput/trend/PercentileThreshold.java
package cz.encircled.jput.trend;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Vlad on 27-May-17.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD})
public @interface PercentileThreshold {

    /**
     * Target percentile (e.g. 70, 90, 95 etc)
     */
    int percentile();

    /**
     * Upper threshold in percents, i.e. allowed variation
     */
    int threshold() default 10;

}

File: jput-core/src/main/java/cz/encircled/jput/trend/PercentileThreshold.kt
package cz.encircled.jput.trend

/**
 * @author Vlad on 27-May-17.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class PercentileThreshold(

        /**
         * Target percentile (e.g. 70, 90, 95 etc)
         */
        val percentile: Int,
        /**
         * Upper threshold in percents, i.e. allowed variation
         */
        val threshold: Int = 10

)
